test/test_input_recipeset.py

```diff
@@ -1390,3 +1390,171 @@ def testNewBehaviour(self):
 
         pkg = self.generate().walkPackagePath("root")
         self.assertTrue(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["prune"])
+
+
+class TestScmDefaults(RecipesTmp, TestCase):
+    """ Test behaviour of pruneImportScm policy"""
+
+    def testScmDefaultsGit(self):
+        """Test scmDefaults for gitscm"""
+        with TemporaryDirectory() as tmp:
+            os.chdir(tmp)
+            os.mkdir("recipes")
+
+            self.writeRecipe("root", """\
+                root: True
+                checkoutSCM:
+                   scm: git
+                   url: foo@bar
+                buildScript: "true"
+                packageScript: "true"
+                """)
+            pkg = self.generate().walkPackagePath("root")
+            self.assertEqual(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["branch"],
+                    "master")
+            self.assertEqual(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["sslVerify"],
+                    None)
+            self.assertEqual(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["shallow"],
+                    None)
+            self.assertEqual(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["singleBranch"],
+                    None)
+            self.assertFalse(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["submodules"])
+            self.assertFalse(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["recurseSubmodules"])
+            self.assertTrue(pkg.getCheckoutStep().getScmList()[0].getProperties(False)["shallowSubmodules"])
+
+            with open("default.yaml", "w") as f:
```

Comment:
Maybe you could add a `writeDefault` (like the existing `writeConfig`) helper in the `RecipesTmp` class?